build failure running wmk.bat -x yard install on Windows 7

- note:  I hit this on a fork of a fork.  I don't think this has been fixed in main, but I'm not 100% sure.  apologies if this is already fixed.

Windows 7 Enterprise
- Get vertx source
- install jython
- set JYTHON_HOME to jython install directory (not the /bin dir)
- open windows cmd shell
- cd to vertx source dir (where wmk.bat is)
- run:  wmk.bat -x yard install

expected
 a successful build

actual
  build fails with

:vertx-lang-jython:pydoc

FAILURE: Build failed with an exception.
- What went wrong:
  Execution failed for task ':vertx-lang-jython:pydoc'.
  > A problem occurred starting process 'command 'C:\programs\jython2.5.3/jython''

Caused by: java.io.IOException: Cannot run program "C:\programs\jython2.5.3/jython" (in directory "C:\Dev\Git\vert.x\vertx-lang\vertx-lang-jython"): CreateProcess error=2
, The system cannot find the file specified

I "fixed" this by modifying

/vertx-lang/vertx-lang-jython/build.gradle

to include the '.bat' file extension to "/jython" @ l.47.  After my "fix", It seems like there's still an error in the build step, but it's non-fatal. Sorry I don't have time to really fix it :(

full stack trace:
- Exception is:
  org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':vertx-lang-jython:pydoc'.
      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:68)
      at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
      at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:34)
      at org.gradle.api.internal.changedetection.CacheLockHandlingTaskExecuter$1.run(CacheLockHandlingTaskExecuter.java:34)
      at org.gradle.cache.internal.DefaultCacheAccess$2.create(DefaultCacheAccess.java:200)
      at org.gradle.cache.internal.DefaultCacheAccess.longRunningOperation(DefaultCacheAccess.java:172)
      at org.gradle.cache.internal.DefaultCacheAccess.longRunningOperation(DefaultCacheAccess.java:198)
      at org.gradle.cache.internal.DefaultPersistentDirectoryStore.longRunningOperation(DefaultPersistentDirectoryStore.java:137)
      at org.gradle.api.internal.changedetection.DefaultTaskArtifactStateCacheAccess.longRunningOperation(DefaultTaskArtifactStateCacheAccess.java:83)
      at org.gradle.api.internal.changedetection.CacheLockHandlingTaskExecuter.execute(CacheLockHandlingTaskExecuter.java:32)
      at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:55)
      at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:57)
      at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:41)
      at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
      at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:52)
      at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:42)
      at org.gradle.api.internal.AbstractTask.executeWithoutThrowingTaskFailure(AbstractTask.java:247)
      at org.gradle.execution.DefaultTaskGraphExecuter.executeTask(DefaultTaskGraphExecuter.java:192)
      at org.gradle.execution.DefaultTaskGraphExecuter.doExecute(DefaultTaskGraphExecuter.java:177)
      at org.gradle.execution.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:83)
      at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:36)
      at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
      at org.gradle.execution.DefaultBuildExecuter.access$200(DefaultBuildExecuter.java:23)
      at org.gradle.execution.DefaultBuildExecuter$2.proceed(DefaultBuildExecuter.java:67)
      at org.gradle.api.internal.changedetection.TaskCacheLockHandlingBuildExecuter$1.run(TaskCacheLockHandlingBuildExecuter.java:31)
      at org.gradle.cache.internal.DefaultCacheAccess$1.create(DefaultCacheAccess.java:111)
      at org.gradle.cache.internal.DefaultCacheAccess.useCache(DefaultCacheAccess.java:126)
      at org.gradle.cache.internal.DefaultCacheAccess.useCache(DefaultCacheAccess.java:109)
      at org.gradle.cache.internal.DefaultPersistentDirectoryStore.useCache(DefaultPersistentDirectoryStore.java:129)
      at org.gradle.api.internal.changedetection.DefaultTaskArtifactStateCacheAccess.useCache(DefaultTaskArtifactStateCacheAccess.java:79)
      at org.gradle.api.internal.changedetection.TaskCacheLockHandlingBuildExecuter.execute(TaskCacheLockHandlingBuildExecuter.java:29)
      at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
      at org.gradle.execution.DefaultBuildExecuter.access$200(DefaultBuildExecuter.java:23)
      at org.gradle.execution.DefaultBuildExecuter$2.proceed(DefaultBuildExecuter.java:67)
      at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
      at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:61)
      at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:54)
      at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:155)
      at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:110)
      at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:78)
      at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:38)
      at org.gradle.launcher.exec.InProcessGradleLauncherActionExecuter.execute(InProcessGradleLauncherActionExecuter.java:39)
      at org.gradle.launcher.exec.InProcessGradleLauncherActionExecuter.execute(InProcessGradleLauncherActionExecuter.java:25)
      at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:50)
      at org.gradle.launcher.cli.ActionAdapter.execute(ActionAdapter.java:30)
      at org.gradle.launcher.cli.ActionAdapter.execute(ActionAdapter.java:22)
      at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:200)
      at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:173)
      at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:169)
      at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:138)
      at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
      at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
      at org.gradle.launcher.Main.doAction(Main.java:48)
      at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
      at org.gradle.launcher.Main.main(Main.java:39)
      at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:50)
      at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:32)
      at org.gradle.launcher.GradleMain.main(GradleMain.java:26)
      at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:33)
      at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:124)
      at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:53)
  Caused by: org.gradle.process.internal.ExecException: A problem occurred starting process 'command 'C:\programs\jython2.5.3/jython''
      at org.gradle.process.internal.DefaultExecHandle.setEndStateInfo(DefaultExecHandle.java:195)
      at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:325)
      at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:90)
      at org.gradle.internal.concurrent.DefaultExecutorFactory$StoppableExecutorImpl$1.run(DefaultExecutorFactory.java:66)
  Caused by: java.io.IOException: Cannot run program "C:\programs\jython2.5.3/jython" (in directory "C:\Dev\Git\vert.x\vertx-lang\vertx-lang-jython"): CreateProcess error=2
  , The system cannot find the file specified
      at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:71)
      ... 1 more
  Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
      ... 2 more

--- a/vertx-lang/vertx-lang-jruby/build.gradle
+++ b/vertx-lang/vertx-lang-jruby/build.gradle
@@ -49,6 +49,10 @@ task yardoc(type:Exec) {
 		"--output-dir","${project.buildDir}/docs/ruby/api",
 		"src/main/ruby/**/*.rb", "src/main/ruby_scripts/**/*.rb",
 		"-", "$rootDir/LICENSE.txt" ]
+		
+  if( System.properties['os.name'].toLowerCase().contains('windows') ) {
+    commandLine = ['cmd', '/c'] + commandLine
+  }
 }
 
 javadoc.dependsOn yardoc

--- a/vertx-lang/vertx-lang-jython/build.gradle
+++ b/vertx-lang/vertx-lang-jython/build.gradle
@@ -45,11 +45,14 @@ task pydoc(type: Exec) {
 	workingDir = project.projectDir
 	
 	commandLine = [ System.getenv()['JYTHON_HOME'] + "/jython",
-		"-J-classpath", project.files(project.configurations.compile).files.join(':'), 
 		"-Dproj.base=${project.projectDir}",
-		"-Dpython.path=${project.projectDir}/src/build_tools/doclib:src/main/python_scripts",
+		"-Dpython.path=${project.projectDir}/src/build_tools/doclib" + System.properties['path.separator'] + "src/main/python_scripts",
 		"-Dproj.base=$project.projectDir.path",
 		"${project.projectDir}/src/build_tools/pydocx.py" ]
+	
+  if( System.properties['os.name'].toLowerCase().contains('windows') ) {
+    commandLine = ['cmd', '/c'] + commandLine
+  }	
 }
 
 javadoc.dependsOn pydoc

